src/cpp/security/accesscontrol/DistinguishedName.h

```diff
@@ -0,0 +1,201 @@
+
+/**
+ * @brief This function compares 2 strings representing DistinguishedNames.
+ *
+ * Follow RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253
+ * The idea here is that comparing 2 DistinguishedNames is not as trivial as comparing 2 strings,
+ * nor comparing its values splitting the string.
+ * The comparison must be a complex function that takes into account many specific details.
+ *
+ * So far, Fast DDS gives 2 strings (in some format that we assume is the correct one) and this compares
+ * whether the two strings refer to the same DistinguishedName.
+ * It would be better (and easier) to store DistinguishedName in a specialized function instead of in a string,
+ * and thus this comparison would be much easier and efficient.
+ * But this will be left for future improvements.
+ *
+ * @warning this function does not correctly fulfilled all requirements of RFC 2253.
+ * Requirements that are not fulfilled:
+ * - multi-attribute
+ * - types accept oid
+ * - assumes that input is a well formed DistinguishedName following rfc2253
+ * @todo make this function complete (probably better by implementing DistinguishedName class).
+ *
+ * @param name1 first DistinguishedName in string format
+ * @param name2 second DistinguishedName in string format
+ *
+ * @return true if both first strings represent the same DistinguishedName.
+ * @return false otherwise
+ */
+bool rfc2253_string_compare(
+        const DistinguishedName& name1,
+        const DistinguishedName& name2);
+
+
+namespace details {
+
+struct Attribute
+{
+    Attribute() = default;
+
+    Attribute(
+            const DistinguishedName& name);
+
+    Attribute(
+            const char* name);
+
+    bool is_set() const noexcept;
+
+    size_t cut(
+            size_t ini,
+            size_t fin = 0) noexcept;
+
+    size_t trim_blank_spaces();
+
+    size_t trim_back_blank_spaces();
+
+    size_t clear() noexcept;
+
+    size_t find(
+            char c) const noexcept;
+
+    size_t find(
+            char c,
+            bool& result) const noexcept;
+
+    bool found(
+            const size_t& find_result) const noexcept;
+
+    bool operator ==(
+            const Attribute& other) const noexcept;
+
+    bool operator !=(
+            const Attribute& other) const noexcept;
+
+    static Attribute cut(
+            const Attribute& att,
+            size_t ini,
+            size_t fin = 0) noexcept;
+
+    const char* value {nullptr};
+    size_t length {0};
+};
+
+/**
+ * @brief This class is only used to compare 2 DistinguishedName.
+ *
+ * @warning be aware that this class does not store any memory, so using for something different than
+ * comparison may lead to seg faults.
+ */
+class DistinguishedNameSpecialized
+{
+public:
+
+    static bool compare(
+            const DistinguishedName& name1,
+            const DistinguishedName& name2) noexcept;
+
+    enum class ErrorCase
+    {
+        ok,
+        empty,
+        max_value,
+        no_type_value_format,
```

Comment:
NIT
To keep uniformity, consider changing it to upper case

Reply:
Hard to say, enum class values are values, not constants. Thus they should be in lower case: https://groups.google.com/a/dartlang.org/g/misc/c/MQzUXwY2rbI
Actually they should be lowerCameCase, but that would change uniformity of variable names.

Reply:
Yeah, it is not so clear,... and those are not constants.... It still makes sense to me just to be coherent with the rest of definitions of this kind